modules/tinymce/src/core/test/ts/browser/fmt/FormatChangeVarsTest.ts

```diff
@@ -0,0 +1,199 @@
+  const browser = PlatformDetection.detect().browser;
+
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Theme ]);
+
+  const events = {
+    general: [] as boolean[],
+    helvetica: [] as boolean[],
+    comicSans: [] as boolean[]
+  };
+
+  const generalCleanup = Singleton.unbindable();
+  const helveticaCleanup = Singleton.unbindable();
+  const comicSansCleanup = Singleton.unbindable();
+
+  const clearEvents = () => {
+    events.general = [];
+    events.helvetica = [];
+    events.comicSans = [];
+  };
+
+  const helveticaFont = 'helvetica, arial, sans-serif';
+  const comicSansFont = '\'comic sans ms\', sans-serif';
+
+  before(() => {
+    const editor = hook.editor();
+    generalCleanup.set(editor.formatter.formatChanged('fontname', (evt) => events.general.push(evt), true));
+    helveticaCleanup.set(editor.formatter.formatChanged('fontname', (evt) => events.helvetica.push(evt), false, {
+      value: helveticaFont
+    }));
+    comicSansCleanup.set(editor.formatter.formatChanged('fontname', (evt) => events.comicSans.push(evt), false, {
+      value: comicSansFont
+    }));
+  });
+
+  after(() => {
+    generalCleanup.clear();
+    helveticaCleanup.clear();
+    comicSansCleanup.clear();
+  });
+
+  context('Simple cases', () => {
+    beforeEach(() => {
+      const editor = hook.editor();
+      editor.setContent(`<p>Unstyled content <span style="font-family: ${comicSansFont}">Styled content</span></p>`);
+      TinySelections.setCursor(editor, [ 0, 0 ], 0);
+      clearEvents();
+    });
+
+    it('TINY-7713: Fires when entering styled content', () => {
+      const editor = hook.editor();
+      // Inside the span
+      TinySelections.setCursor(editor, [ 0, 1, 0 ], 0);
+
+      assert.deepEqual(events.general, [ true ], 'font-family has been turned on');
+      assert.deepEqual(events.comicSans, [ true ], 'font-family comic sans has been turned on');
+    });
+
+    it('TINY-7713: Fires when leaving styled content', () => {
+      const editor = hook.editor();
+      // Move into the styled area
+      TinySelections.setCursor(editor, [ 0, 1, 0 ], 0);
+      clearEvents();
+      // Move back out
+      TinySelections.setCursor(editor, [ 0, 0 ], 0);
+
+      assert.deepEqual(events.general, [ false ], 'font-family was turned off');
+      assert.deepEqual(events.comicSans, [ false ], 'font-family comic sans was turned off');
+    });
+
+    it('TINY-7713: Fires when changing formats', () => {
+      const editor = hook.editor();
+      // Select the entire styled content
+      TinySelections.setSelection(editor, [ 0, 1, 0 ], 0, [ 0, 1, 0 ], 'Styled content'.length);
+      clearEvents();
+      // Change it
+      editor.formatter.apply('fontname', { value: helveticaFont });
+
+      if (browser.isIE) {
+        assert.isTrue(Arr.last(events.general).forall(Fun.identity), 'font-family was either unchanged, or ended up in the on state');
```

Comment:
- [ ] See discussion in slack, can we fix whatever is causing IE to behave like this?

Reply:
Not going to fix this, as it always ends up in the right state anyway and was an existing issue